// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------


syntax = "proto3";


message ProposeUpdateUserManager {

    string user_id = 1;

    string new_manager_id = 2;

    string reason = 3;

    string metadata = 4;

}


message ConfirmUpdateUserManager {

    string proposal_id = 1;

    string reason = 2;

}


message RejectUpdateUserManager {

    string proposal_id = 1;

    string reason = 2;

}


message CreateUser {

    string name = 1;

    string user_id = 2;

    string manager_id = 3;

    string metadata = 4;

}


message UpdateUser {

    string user_id = 1;

    string new_name = 2;

    string old_metadata_sha512 = 3;

    string new_metadata = 4;

}
